ibm/provider/provider.go

```diff
@@ -893,6 +893,7 @@ func Provider() *schema.Provider {
 			"ibm_cos_bucket":                            cos.ResourceIBMCOSBucket(),
 			"ibm_cos_bucket_replication_rule":           cos.ResourceIBMCOSBucketReplicationConfiguration(),
 			"ibm_cos_bucket_object":                     cos.ResourceIBMCOSBucketObject(),
+			"ibm_cos_bucket_objectlock_configuration":   cos.ResourceIBMCOSBucketObjectlock(),
```

Comment:
ibm_cos_bucket_objectlock_configuration to ibm_cos_bucket_object_lock_configuration